### PR TITLE
Adding Raysect

### PR DESCRIPTION
**Pull request by munechika-koyo**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by munechika-koyo** [comment]
@CnlPepper,  @mattngc,
I would like to add `raysect` to the conda-forge community package.
Please comment on this PR because your comments are required to approve this PR.

File: recipes/raysect/recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    - ${{ stdlib("c") }}
+  host:
+    - python
+    - pip
+    - setuptools
+    - cython <3.0
+    - numpy <2.0
+  run:
+    - python
+    - numpy <2.0
```
Comment (xhochy) [inline recipes/raysect/recipe.yaml:36]: If numpy is in host, the run dependency comes through run_exports 
```suggestion
```

File: recipes/raysect/recipe.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    - if: win
+      then:
+        - ${{ compiler("clang") }}
+    - ${{ stdlib("c") }}
+  host:
+    - python
+    - pip
+    - setuptools
+    - cython <3.0
+    - numpy <2.0
```
Comment (xhochy) [inline recipes/raysect/recipe.yaml:33]: This should use the global pin
```suggestion
    - numpy
```

File: recipes/raysect/recipe.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    - win
+
+requirements:
+  build:
+    - if: unix
+      then:
+        - ${{ compiler("c") }}
+    - if: win
+      then:
+        - ${{ compiler("clang") }}
```
Comment (xhochy) [inline recipes/raysect/recipe.yaml:26]: Please use a variant file to override the compiler.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/raysect/recipe.yaml (hunk 1 above)
Comment (munechika-koyo) [reply recipes/raysect/recipe.yaml:36]: Did you mention `pin_compatible("numpy")`?

Comment (munechika-koyo) [reply recipes/raysect/recipe.yaml:36]: Sorry I am confused.
what you mentined is that we should put same `numpy` in the run dependency in order to appropriately link against `numpy` in the host environment, right?
So, should I replace `numpy <2.0` into `${{ pin_subpackage("numpy", exact=True) }}` or the following expression?

```yaml
run_exports:
    - ${{ pin_subpackage("numpy", exact=True) }}
```

If I tried to opt for the latter, the following error was occurred:
`Error:   × Missing output: numpy (used in pin_subpackage)`

Comment (xhochy) [reply recipes/raysect/recipe.yaml:36]: No, I want you to exactly do what I suggested, delete `numpy` completely from the explicitly listed run requirements.

Comment (munechika-koyo) [reply recipes/raysect/recipe.yaml:36]: Ah, so if `numpy` is specified in host deps, it is automatically tied to run requirement through `run_export`.
I got it.

File: recipes/raysect/recipe.yaml (hunk 2 above)
Comment (munechika-koyo) [reply recipes/raysect/recipe.yaml:33]: Should I remove the version specification?

Comment (xhochy) [reply recipes/raysect/recipe.yaml:28]: Yes

Comment (munechika-koyo) [reply recipes/raysect/recipe.yaml]: But, how should I set the `numpy <2.0` limitation in runtime?

Comment (munechika-koyo) [reply recipes/raysect/recipe.yaml]: Ah, yes, I can see it!
Additionally, at the default [`conda_build_config.yaml` ](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml), `numpy` version is specified at each python one.

**Comment by munechika-koyo** [comment]
@conda-forge/staged-recipes this is ready to merge.

**Comment by munechika-koyo** [comment]
@conda-forge/help-python-c

**Review by xhochy — APPROVED** [review]

**Pull request merged**